app/models/manageiq/providers/nsxt/inventory/parser/network_manager.rb

```diff
@@ -48,81 +53,122 @@ def network_routers
 
   def cloud_networks
     collector.segments.each do |segment|
+      next if segment['type'] == 'DISCONNECTED'
       cloud_network = persister.cloud_networks.find_or_build(segment['id'])
       cloud_network.name = segment['display_name']
       cloud_network.description = segment['description']
       cloud_network.cloud_tenant = cloud_tenant(segment['tags'])
       cloud_network.status = 'active'
       cloud_network.enabled = true
-      cloud_network.extra_attributes = { :security_zone => get_tag_value_by_scope(segment['tags'], 'network_zone') }
-      cloud_subnets(segment)
+      network_router_id = segment['connectivity_path']&.split('/tier-1s/')&.last
+      network_router = persister.network_routers.lazy_find(network_router_id) unless network_router_id.nil?
+      cloud_subnets(segment, network_router)
```

Comment:
If `network_router_id.nil?` we probably should skip `cloud_subnets(segment, network_router)` also since it won't have all of the keys necessary for its manager_ref and save_inventory will fail